src/firecrest/dependencies.py

```diff
@@ -300,10 +300,27 @@ async def __call__(
         system = ServiceAvailabilityDependency(
             service_type=HealthCheckType.scheduler, ignore_health=self.ignore_health
         )(system_name=system_name)
+
+        is_ssh_healthy = False
+        requires_ssh_healthy = not self.ignore_health
+
+        if requires_ssh_healthy:
```

Comment:
Why are you adding this health verification? I am not convinced that removing SSH when it is not healthy is the correct approach. You are introducing some side effects. For example if a cluster is configured to use ssh as a fallback to the API we don't want to exclude the ssh client based on its health.

To decide if the ssh client needs to be provided or not, we should look at the cluster configuration. If SSH is configured, then we should always provide the ssh client.